packages/sui-segment-wrapper/src/repositories/googleRepository.js

```diff
@@ -48,14 +46,38 @@ export const loadGoogleAnalytics = async () => {
   return loadScript(gtagScript)
 }
 
+// Trigger GA init event just once per session.
+const triggerGoogleAnalyticsInitEvent = sessionId => {
+  const eventName = getConfig('googleAnalyticsInitEvent') ?? DEFAULT_GA_INIT_EVENT
+  const eventPrefix = `ga_event_${eventName}_`
+  const eventKey = `${eventPrefix}${sessionId}`
+
+  // Check if the event has already been sent in this session.
+  if (!localStorage.getItem(eventKey)) {
+    // If not, send it.
+    window.gtag('event', eventName)
+    console.log(`Sending GA4 event "${eventName}" for the session "${sessionId}"`)
+
+    // And then save a new GA session hit in local storage.
+    localStorage.setItem(eventKey, 'true')
+  }
+
+  // Clean old GA sessions hits from the storage.
+  Object.keys(localStorage).forEach(key => {
+    if (key.startsWith(eventPrefix) && key !== eventKey) {
+      localStorage.removeItem(key)
+    }
+  })
+}
+
 const getGoogleField = async field => {
   const googleAnalyticsMeasurementId = getConfig('googleAnalyticsMeasurementId')
 
-  // If `googleAnalyticsMeasurementId` is not present, don't load anything
+  // If `googleAnalyticsMeasurementId` is not present, don't load anything.
```

Comment:
👏

Reply:
Sorry, but I'm very pro to treat comments as sentences 🙏  https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html